### PR TITLE
Add half-life theme inspired from oh-my-zsh

### PR DESCRIPTION
**Pull request by memduhcagridemir**
This PR adds a new theme similar to half-life theme of oh-my-zsh.

Here is a screenshot of how it looks;

half-life theme

### CODE REVIEW

**Comment by memduhcagridemir** [comment]
@nntoan @chaifeng could you take a look?

File: themes/half-life/half-life.theme.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  CHAR=$(scm_char)
+  if [ $CHAR = $SCM_NONE_CHAR ]
+    then
+      return
+    else
+      echo " on ${blue}$(git_current_branch)$(parse_git_dirty)${normal} "
+  fi
+}
+
+function prompt_command() {
```
Comment (akinomyoga) [inline themes/half-life/half-life.theme.sh:17]: It is better to make these two functions consistent.

```suggestion
prompt_command() {
```

File: themes/half-life/half-life.theme.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+OSH_THEME_GIT_PROMPT_DIRTY="✗"
+OSH_THEME_GIT_PROMPT_CLEAN="✓"
+
+# Nicely formatted terminal prompt
+ps_scm_prompt() {
+  CHAR=$(scm_char)
```
Comment (akinomyoga) [inline themes/half-life/half-life.theme.sh:8]:
```suggestion
  local CHAR=$(scm_char)
```

File: themes/half-life/half-life.theme.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      return
+    else
+      echo " on ${blue}$(git_current_branch)$(parse_git_dirty)${normal} "
+  fi
+}
+
+function prompt_command() {
+  ps_username="${purple}\u${normal}"
+  ps_path="${green}\w${normal}"
+  ps_user_mark="${orange}λ${normal}"
```
Comment (akinomyoga) [inline themes/half-life/half-life.theme.sh:20]:
```suggestion
  local ps_username="${purple}\u${normal}"
  local ps_path="${green}\w${normal}"
  local ps_user_mark="${orange}λ${normal}"
```

File: themes/half-life/half-life.theme.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+OSH_THEME_GIT_PROMPT_DIRTY="✗"
+OSH_THEME_GIT_PROMPT_CLEAN="✓"
+
+# Nicely formatted terminal prompt
+ps_scm_prompt() {
+  CHAR=$(scm_char)
+  if [ $CHAR = $SCM_NONE_CHAR ]
```
Comment (akinomyoga) [inline themes/half-life/half-life.theme.sh:9]:
```suggestion
  if [ "$CHAR" = "$SCM_NONE_CHAR" ]
```

**Review by akinomyoga — CHANGES_REQUESTED** [review]

**Comment by akinomyoga** [comment]
Ah, sorry. As you have been working on the `master` branch for this PR, I have failed to correctly push the rebased commits. The PR has been unintendedly closed, and I cannot reopen the PR again...

I'm sorry, but could you again force-push your changes to the `master` branch of `memduhcagridemir/oh-my-bash`? Or, can you force-push the commit of the [`pull255`](https://github.com/ohmybash/oh-my-bash/tree/pull255) branch of `ohmybash/oh-my-bash` to the `master` branch of `memduhcagridemir/oh-my-bash`?

<details><summary><b>In case you don't know how to do it, let me explain it here.</b></summary>

**Step 1**: If you don't have a local repository, you first need to clone the GitHub repository to a local repository:

```bash
$ git clone git@github.com:memduhcagridemir/oh-my-bash.git
$ cd oh-my-bash
$ git remote add upstream git@github.com:akinomyoga/oh-my-bash.git
```

**Step 2**: Check the names of remote repositories inside the local repository

```bash
$ git remote -v
origin        git@github.com:memduhcagridemir/oh-my-bash (fetch)
origin        git@github.com:memduhcagridemir/oh-my-bash (push)
upstream        git@github.com:ohmybash/oh-my-bash (fetch)
upstream        git@github.com:ohmybash/oh-my-bash (push)
```

In this case, your GitHub repository is `origin` and the source repository is `upstream`. If you haven't registered the source repository, you need add it by

```bash
$ git remote add upstream git@github.com:akinomyoga/oh-my-bash.git
```

**Step 3**: If you have any unsaved changes in the working tree or in the index, please save them (by making commits or by `git stash`).

**Step 4**: Please replace `upstream` in the following command with the name that we have checked in step 2.

```bash
$ git checkout master
$ git branch master-backup
$ git fetch upstream
$ git reset --hard upstream/master
$ git push -f origin master
```

</details>

**Pull request closed without merging**